extra_data/file_access.py

```diff
@@ -467,6 +467,21 @@ def add_key(key, value):
         self._keys_cache[source] = res
         return res
 
+    def get_one_key(self, source):
```

Comment:
We also have two pieces of caching we could use in `FileAccess` to potentially avoid accessing the file. `fa._keys_cache` stores complete sets of keys as returned by `fa.keys()`, and `fa._known_keys` stores partial sets as used by `key in source`.

It won't make a difference for `plot_missing_data()` as you're only looking at each source once, but it's probably worth using, whether we add it in this PR or separately.

Reply:
Good point, fixed in 029011e.